connectors/src/connectors/confluence/temporal/activities.ts

```diff
@@ -439,3 +439,78 @@ export async function confluenceRemoveSpaceActivity(
     await deletePage(connectorId, page.pageId, dataSourceConfig);
   }
 }
+
+export async function fetchConfluenceSpaceIdsForConnectorActivity({
+  connectorId,
+}: {
+  connectorId: ModelId;
+}) {
+  const spacesForConnector = await ConfluenceSpace.findAll({
+    attributes: ["spaceId"],
+    where: {
+      connectorId,
+    },
+  });
+
+  return spacesForConnector.map((s) => s.spaceId);
+}
+
+// Personal Data Reporting logic.
+
+interface ConfluenceUserAccountAndConnectorId {
+  connectorId: ModelId;
+  userAccountId: string;
+}
+
+export async function fetchConfluenceUserAccountAndConnectorIdsActivity(): Promise<
+  ConfluenceUserAccountAndConnectorId[]
+> {
+  return ConfluenceConfiguration.findAll({
+    attributes: ["connectorId", "userAccountId"],
+  });
+}
+
+export async function confluenceGetReportPersonalActionActivity(
+  params: ConfluenceUserAccountAndConnectorId
+) {
+  const { connectorId, userAccountId } = params;
+
+  const connector = await Connector.findOne({
+    attributes: ["connectionId"],
+    where: {
+      id: connectorId,
+    },
+  });
+  if (!connector) {
+    throw new Error(`Connector not found (connectorId: ${connectorId})`);
+  }
+
+  // We look for the oldest updated data.
+  const oldestPageSync = await ConfluencePage.findOne({
+    where: {
+      connectorId,
+    },
+    order: [["lastVisitedAt", "ASC"]],
```

Comment:
We likely need an index here?

Reply:
Yep, I plan to add one in another PR.

Reply:
Added in https://github.com/dust-tt/dust/pull/3418.

Reply:
✅